ouroboros-network-api/src/Ouroboros/Network/PeerSelection/LedgerPeers/Type.hs

```diff
@@ -85,3 +212,40 @@ data LedgerPeersConsensusInterface m = LedgerPeersConsensusInterface {
     lpGetLedgerStateJudgement :: STM m LedgerStateJudgement,
     lpGetLedgerPeers          :: STM m [(PoolStake, NonEmpty RelayAccessPoint)]
   }
+
+instance ToJSON RelayAccessPointCoded where
+  toJSON (RelayAccessPointCoded (RelayAccessDomain domain port)) =
+    object
+      [ "domain" .= fullyQualified
+      , "port"   .= (fromIntegral port :: Int)]
+    where
+      fullyQualified =
+        case BS.unsnoc domain of
+          Just (_, '.') -> BS.unpack domain
+          Just _        -> BS.unpack $ BS.snoc domain '.'
+          Nothing       -> mempty
+
+  toJSON (RelayAccessPointCoded (RelayAccessAddress ip port)) =
+    object
+      [ "address" .= show ip
+      , "port" .= (fromIntegral port :: Int)]
+
+instance FromJSON RelayAccessPointCoded where
+  parseJSON = withObject "RelayAccessPointCoded" $ \v -> do
+    domain <- fmap BS.pack <$> v .:? "domain"
+    port <- fromIntegral @Int <$> v .: "port"
+    case domain of
+      Nothing ->
+            v .: "address"
+        >>= \case
+               Nothing -> fail "RelayAccessPointCoded: invalid IP address"
+               Just addr ->
+                 return . RelayAccessPointCoded $ RelayAccessAddress addr port
+            . readMaybe
+
+      Just domain'
+        | Just (_, '.') <- BS.unsnoc domain' ->
+          return . RelayAccessPointCoded $ RelayAccessDomain domain' port
+        | otherwise ->
+          let fullyQualified = domain' `BS.snoc` '.'
+          in return . RelayAccessPointCoded $ RelayAccessDomain fullyQualified port
```

Comment:
So this is only to not parse `1.1.1.1.` as an `IP` address.  For that we'd also need a newer version of `iproute`.

I  would avoid this at all. This should be in the ledger rules.

Reply:
No, IP addresses are handled by the domain constructor. This is a check to ensure that domain name we have read is fully qualified, and if not, add a period at the end. I realize now that this is redundant since I serialise all domain names in fully qualified format - maybe I wrote the former before I added the latter.